File: LeetCode/ProblemSet/Volume (3700-3799)/LeetCode3737/LeetCode3737-Solution-01.cpp
#include <bits/stdc++.h>

class Solution {
public:
    int countMajoritySubarrays(std::vector<int>& nums, int target) {
       const int n = (int) nums.size();
       
        int ans = 0;
        for (int i = 0; i < n; i++) {
            int cnt = 0;
            for (int j = i; j < n; j++) {
                nums[j] == target ? cnt++ : cnt--;
                if (cnt > 0) {
                    ans++;
                }
            }
        }

        return ans;
    }
};

File: LeetCode/ProblemSet/Volume (3700-3799)/LeetCode3737/LeetCode3737-Solution-01.go
package leetcode3737

func countMajoritySubarrays(nums []int, target int) int {
	n, ans := len(nums), 0
	for i := 0; i < n; i++ {
		cnt := 0
		for j := i; j < n; j++ {
			if nums[j] == target {
				cnt++
			} else {
				cnt-- 
			}

			if cnt > 0 {
				ans++
			}
		}
	}

	return ans
}